kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -191,32 +191,41 @@ public class TableConfig<T> {
   ///////////////////////////
 
   /**
-   * Validates that the given properties have the delta prefix in the key name, and they are in the
-   * set of valid properties. The caller should get the validated configurations and store the case
-   * of the property name defined in TableConfig.
+   * Validates that the given new properties that the txn is trying to update in table. Properties
+   * that have `delta.` prefix in the key name should be in valid list and are editable. The caller
+   * is expected to store the returned properties in the table metadata after further validation
+   * from a protocol point of view. The returned properties will have the key's case normalized as
+   * defined in its {@link TableConfig}.
```

Comment:
Should a test check that the keys are normalized? it seems like no configs are not editable currently so we cannot verify that it seems

Reply:
good point. Adding a test. We have editable properties such as checkpointInterval